dashboard/utils.py

```diff
@@ -241,25 +242,66 @@ def get_payments_count_for_course(self, course):
             course_key__in=course.courserun_set.values('edx_course_key'),
         ).values('order_id').distinct().count()
 
-    def get_custom_number_of_attempts_for_course(self, course):
+    def get_number_of_attempts_left(self, course):
```

Comment:
I'm sorry if I'm missing something basic but I have a confusion here for a case with the new Logic:

A user placing an Order before the first date and then using both his attempts before the first date.

Now the user places another order after the second date as a scenario mentioned in [this doc](https://docs.google.com/document/d/1A_fZemxKx6haTJj0euWgaZXjFvcsWX7RgP6cIJR8XbY). With the new order shouldn't it return at least 1 attempt?
This is related to how/when we create the `ExamAuthorization` for an Order, I believe that part is [here](https://github.com/mitodl/micromasters/blob/master/exams/api.py#L84-L89). Looking at this will it create the new ExamAuthorization for a New Order with our updated logic?

Reply:
Yes, if a learner had a payment and two attempts before the first date, if they make another purchase they get another exam attempt and has_to_pay_for_exam would return False.

Reply:
Thanks for adding the details. Just let me know when the PR is ready for review again